project_budget_management/models/project_project.py

```diff
@@ -0,0 +1,183 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from datetime import timedelta
+
+from odoo import api, fields, models
+
+
+class ProjectProjectStage(models.Model):
+    _inherit = 'project.project.stage'
+
+    is_close = fields.Boolean("Closing Stage")
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    def _compute_spent_budget(self):
+        """
+        Compute the Spent Budget of Project
+        """
+        timesheet_invoice_obj = self.env["timesheet.invoice"]
+        for project in self:
+            timesheet_invoice_records = timesheet_invoice_obj.sudo().search(
+                [
+                    ("project_id", "=", project.id),
+                    ("state", "in", ("approved", "completed")),
+                ]
+            )
+            project.spent_budget = sum(
+                inv.total_amount for inv in timesheet_invoice_records
+            )
+
+    def _compute_running_cost(self):
+        """
+        Compute Running Cost of Project.
+        """
+        timesheet_invoice_obj = self.env["timesheet.invoice"]
+        for project in self:
+            timesheet_invoice_records = timesheet_invoice_obj.search(
+                [
+                    ("project_id", "=", project.id),
+                    ("state", "in", ("draft", "confirm", "pre-approved")),
+                ]
+            )
+            project.running_cost = sum(
+                inv.total_amount for inv in timesheet_invoice_records
+            )
+
+    @api.depends("actual_budget", "revised_budget")
+    def _compute_actual_revised_budget(self):
+        for project in self:
+            actual_revised_budget = project.actual_budget
+            if project.revised_budget > 0:
+                actual_revised_budget = project.revised_budget
+            project.actual_revised_budget = actual_revised_budget
+
+    @api.depends("actual_budget", "spent_budget", "percentage_completed")
+    def _compute_budget_completion(self):
+        for project in self:
+            budget_of_completion = 0.0
+            if project.percentage_completed > 0 and project.stage_id.is_close is False:
+                budget_of_completion = (
+                    (
+                        (100 - project.percentage_completed)
+                        / project.percentage_completed
+                    )
+                    * project.spent_budget
+                ) + project.spent_budget
+            project.budget_of_completion = budget_of_completion
+
+    def _compute_assignee_id_editable(self):
+        cann_pro_crd = self.env.user.has_group(
+            "project_management_security.group_project_coordinator"
```

Comment:
dependency for project_management_security module is missing in manifest

Reply:
module is indirectly depends on project_management_security